### PR TITLE
adding location widgets

### PR DESCRIPTION
**Pull request by adamjohnwright**
This pull request contains location widgets for:
- cds
- clone
- feature
- operon
- pcr_product (commented out)
- protein
- pseudogene
- sequence
- transcript
- transposon
- variation

The pcr_product widget is not being released because I noticed and error when running the following command:

```bash
curl -g -H content-type:application/json http://localhost:8130/rest/widget/pcr_oligo/cenix:10-a2/location | jq '.
```
Basically my code requires the species field but for this entity it was not available. 

I realise that for protein there still requires some work. I will be doing one more pass through tomorrow

### CODE REVIEW

**Comment by adamjohnwright** [comment]
Just went through this today. The main work that was required was to get it working correctly for protein. I think I am now replicating all the logic in the Perl code. 

The plan for testing is to get this in to staging and then get someone like Scott Cain to look it over.

So far I have gone through several endpoints for each entity type and everything looks good.

**Review by sibyl229 — APPROVED** [review]
I think it's okay for merging. But I will add a bit of my thoughts on the implementation of the genetic-position function. 

The function becomes more complex and opaque, in attempt to handle different input types (such as variations and proteins). If we encounter similar situation in the future, I think maybe we could consider providing multiple implementation of the same function based on input type.

Another more subjective opinion is that dynamically constructed keywords makes code harder to read. For example
```
+                     (let [kw-imp (keyword entity-role str-imp)
 +                           str-imp-component (str entity-role "." str-imp)
 +                           kw-map (keyword entity-role "map")
 +                           kw-map-map (keyword (str entity-role ".map") "map")
 +                           kw-imp-map (keyword str-imp-component "map")
 +                           kw-imp-position (keyword
 +                                             str-imp-component
 +                                             (if (= role "sequence") "float" "position"))]
```
vs 
```
:variation/interpolated-map-position
:varitation.interpolated-map-position/map
```

File: src/rest_api/classes/expression_cluster/widgets/overview.clj [hunk 1]
```diff
@@ -19,8 +19,7 @@
 	      (let [id-kw (first (filter #(= (name %) "id") (keys object)))]
 		{(let [i (id-kw object)]
 		   (if-let [result (re-find #":(\d+)_min" i)]
-		     (do (println (second result))
-			 (pad-zero (second result) 10)) i))
+			 (pad-zero (second result) 10)) i)
```
Comment (mgrbyte) [inline src/rest_api/classes/expression_cluster/widgets/overview.clj:22]: this needs dedenting

**Review by mgrbyte — APPROVED** [review]
Generally, It looks good to merge.

Agree with comments @sibyl229 made about the dynamically constructed keywords (albeit I'm probably guilty of introducing that pattern!). 
The above (dynamic kw construction) could be discussed further, I think it's combination of the nature of the dynamic schema conversion + clojure "foo" (e.g perhaps we could look further into clojure multimethods/protocols vs dynamically handling things based on a "role" string) for this type of generic dispatch.

**Comment by adamjohnwright** [comment]
BTW. I agree that over time we should be coming up with better patterns for doing these things. I would be more than willing to make better code. I am not sure how / when we should do this but we should keep this in the back of our minds.

**Pull request merged**